openedx/frontend-app-authoring
# feat: add an allowlist of for supported blocks in library [FC-0062]

## Description

Libraries v2 currently don't support editing blocks other than problem,
text and videos. This commit adds a configuration variable called
`LIBRARY_SUPPORTED_BLOCKS` to setup allowed list of block types users
can paste into libraries. By default it is set to support
'problem,text,video,html`.

Also, parse and display error message from server on paste from clipboard failure.

~Depends on: https://github.com/openedx/edx-platform/pull/35633~

## Supporting information

* https://github.com/openedx/frontend-app-authoring/issues/1311
* ~https://github.com/openedx/edx-platform/pull/35633~
* `Private-ref`: https://tasks.opencraft.com/browse/FAL-3860

## Testing instructions

* Copy any unsupported block from course which is not part of `LIBRARY_SUPPORTED_BLOCKS` setting in https://github.com/openedx/frontend-app-authoring/pull/1378/files#diff-ec767e14d91498b5b31a4ce826c4596061eb83a47d3dee00f6cbe1078232a75cR136
* Try to paste it in the library, it should display an error message.
